File: esthesis-platform-backend/esthesis-platform-backend-server/src/main/java/esthesis/platform/backend/server/dto/CampaignDTO.java
package esthesis.platform.backend.server.dto;

import esthesis.platform.backend.common.dto.BaseDTO;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

import java.time.Instant;
import java.util.Collection;

@Data
@NoArgsConstructor
@Accessors(chain = true)
@EqualsAndHashCode(callSuper = true)
public class CampaignDTO extends BaseDTO {
  private String name;
  private Integer type;
  private Integer state;
  private String description;
  private String commandName;
  private String commandArguments;
  private String provisioningPackageId;
  private Collection<CampaignConditionDTO> conditions;
  private Collection<CampaignMemberDTO> members;
  private Instant startedOn;
  private Instant terminatedOn;
  private Long provisioningId;
}

File: esthesis-ui/src/app/dto/campaign-dto.ts
import {CampaignConditionDto} from './campaign-condition-dto';
import {CampaignMemberDto} from './campaign-member-dto';

export interface CampaignDto {
  id: number;
  name: string;
  description: string;
  scheduleDate: string;
  scheduleHour: number;
  scheduleMinute: number;
  type: number;
  commandName: string;
  commandArguments: string;
  provisioningPackageId: string;
  conditions: CampaignConditionDto[];
  members: CampaignMemberDto[];
  startedOn: Date;
  terminatedOn: Date;
}
